arthur/tasks.py

```diff
@@ -172,48 +185,59 @@ def add(self, task_id, backend, category, backend_args,
 
         :returns: the new task added to the registry
 
-        :raises AlreadyExistsError: raised when the given task identifier
-            exists on the registry
+        :raises: AlreadyExistsError: raised when the given task identifier
+            already exists in the registry
         """
         self._rwlock.writer_acquire()
-
-        if task_id in self._tasks:
+        try:
+            task_key = self._task_key(task_id)
+            found = self.conn.exists(task_key)
+
+            if found:
+                self._rwlock.writer_release()
+                raise AlreadyExistsError(element=str(task_id))
+
+            task = Task(task_id, backend, category, backend_args,
+                        archiving_cfg=archiving_cfg,
+                        scheduling_cfg=scheduling_cfg)
+            self.conn.set(task_key, pickle.dumps(task))
+
+            logger.debug("Task %s added to the registry", str(task_id))
+            return task
+        except RedisError as e:
+            logger.error("Task %s not added: %s", str(task_id), str(e))
```

Comment:
In the case, this fails we are not managing in any way in the rest of the code this failure. It happens the same with the rest of calls to the registry. What should be the best method? We can either raise a custom exception catch it in where the module calls this or return `False`, `None`, and manage those values too.

Reply:
I'm fine with any solution. If you decide to go for the custom exception, propose its name.